src/util/string.cpp

```diff
@@ -743,3 +743,161 @@ void str_replace(std::string &str, char from, char to)
 {
 	std::replace(str.begin(), str.end(), from, to);
 }
+
+std::wstring translate_all(const std::wstring &s, size_t &i);
+
+std::wstring translate_string(const std::wstring &s, size_t &i) {
+	std::wstring output;
+	std::vector<std::wstring> args;
+	int arg_number = 1;
+	while (i < s.length()) {
+		if (s[i] != '\x1b') {
+			output += s[i];
+			if (s[i] == '@') {
+				output += '@';
+			}
+			++i;
+			continue;
+		}
+		//size_t escape_start = i;
+		++i;
+		size_t start_index = i;
+		size_t length;
+		if (i == s.length()) {
+			length = 0;
+		} else if (s[i] == L'(') {
+			++i;
+			++start_index;
+			while (i < s.length() && s[i] != L')') {
```

Comment:
There's a multi-line comment with enough information already. No need to say what each line does.